sources/imdsclient/src/lib.rs

```diff
@@ -168,6 +168,12 @@ impl ImdsClient {
         self.fetch_string(&instance_type_target).await
     }
 
+    /// Get lifecycle state from instance metadata.
+    pub async fn fetch_lifecycle_state(&mut self) -> Result<Option<String>> {
```

Comment:
Are there other lifecycles this could conflict with? Should this be `fetch_autoscaling_lifecycle_state` instead?